core/src/main/scala/aima/core/search/adversarial/MinimaxSearch.scala

```diff
@@ -0,0 +1,44 @@
+package aima.core.search.adversarial
+
+import aima.core.search.adversarial.{Game, UtilityValue}
+import aima.core.agent.{Action, NoAction}
+
+/**
+  * @author Aditya Lahiri
+  * @author Shawn Garner
+  */
+
+object MinimaxDecision {
+    def minMaxDecision[S, A, P](g: Game[S][A][P]): S, P => A = {
+        (s: S, p: P) => maxMinValue(g.getActions(s), p)
+            
+        def maxMinValue(actions: List[Action], p: P): Action = actions match {
+            case Nil => NoAction
```

Comment:
You'll not want to use `NoAction` but instead pass into minMaxDecision a parameter of type `ACTION` (you currently have as A) 

eg
`def minMaxDecision[S, A, P](g: Game[S][A][P], noAction: A)`